clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt

```diff
@@ -44,12 +60,58 @@ interface FossIdRestService {
         /**
          * The mapper for JSON (de-)serialization used by this service.
          */
-        val JSON_MAPPER = JsonMapper()
+        val JSON_MAPPER: ObjectMapper = JsonMapper()
             .setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE)
             // FossID has a bug in get_results/scan.
             // Sometimes the match_type is "ignored", sometimes it is "Ignored".
             .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS)
-            .registerKotlinModule()
+            .registerModule(
+                kotlinModule().setDeserializerModifier(FossIdDeserializerModifier())
+            )
+
+        /**
+         * A class to modify the standard Jackson deserialization to deal with inconsistencies in responses
+         * sent by the FossID server.
+         * FossID usually returns data as a List or Map, but in case of no entries it returns a Boolean (which is set to
+         * false). To handle this special case, this class makes sure that a custom deserializer is used for lists or
+         * maps that detects such a result and converts it to an empty list or map, respective. That way, the FossID
+         * service interface can use meaningful result types.
+         */
+        private class FossIdDeserializerModifier : BeanDeserializerModifier() {
+            override fun modifyMapDeserializer(
+                config: DeserializationConfig,
+                type: MapType,
+                beanDesc: BeanDescription,
+                deserializer: JsonDeserializer<*>
+            ): JsonDeserializer<*> {
+                return FalseValueDeserializer(deserializer) { mutableMapOf<Any, Any>() }
+            }
```

Comment:
Nit: Could be

            ): JsonDeserializer<*> =
                FalseValueDeserializer(deserializer) { mutableMapOf<Any, Any>() }